<a href="https://colab.research.google.com/github/ashpou1/data-processing-4313/blob/main/SQL%20and%20BigQuery.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#ASHOK SHARMA POUDEL  
#218380  
#DATA PROCESSING VISUALIZATION  4313 
#DR. NICHOLAS JACOB

##Learning Some SQL with BigQuery
The first bit below comes directly from Google, you'll need to do each of those to be successful in getting this document to work.

##Cloud Resource Manager to Create a Cloud Platform project
##Enable BigQuery APIs for the project.

In [1]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


I will use austin_bikeshare data set of google trends.

## TOTAL ROWS
I will use my project id and count my total rows in data frame of bikeshare trips

In [8]:
%%bigquery --project data-viz-338818
SELECT 
  COUNT(*) as total_rows
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

total_rows
0     1424786

$1424786$ is a big data. 

##LONGEST TRIP IN MINUTES
Lets find out longest trip in data set

In [9]:
%%bigquery --project data-viz-338818
SELECT 
  MAX(duration_minutes) as Longest_trip_in_minutes
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

Longest_trip_in_minutes
0                    34238

longest trip in data is $34238$ minutes

##AVERAGE TIME FOR TRIP
lets compute average time for trip in data set. we will use start station name to end station name and duration as minutes. 

In [10]:
%%bigquery --project data-viz-338818
SELECT 
  start_station_name, 
  end_station_name,
  AVG(duration_minutes) as Average_time_for_trip
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name is not null
GROUP BY start_station_name, end_station_name

start_station_name  ... Average_time_for_trip
0                                       Zilker Park West  ...             22.636364
1                                Toomey Rd @ South Lamar  ...             48.985782
2                                       Waller & 6th St.  ...             38.334477
3                                             Pease Park  ...             30.497006
4                                       Waller & 6th St.  ...             18.060000
...                                                  ...  ...                   ...
13962  South Congress & Barton Springs at the Austin ...  ...             29.971429
13963  South Congress & Barton Springs at the Austin ...  ...             49.000000
13964  South Congress & Barton Springs at the Austin ...  ...             39.500000
13965  South Congress & Barton Springs at the Austin ...  ...             42.642857
13966  South Congress & Barton Springs at the Austin ...  ...             47.509804

[13967 rows x 3 columns]

We got as many as 13967 averege average trips which starts from  $22.636364$ to $47.509804$